core/templates/components/profile-link-directives/profile-link-image.component.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for creating image links to a user's profile page.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+
+import { AppConstants } from 'app.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+interface IProfileResponse {
+  'profile_picture_data_url_for_username': string;
+}
+
+@Component({
+  selector: 'profile-link-image',
+  templateUrl: './profile-link-image.component.html',
+  styleUrls: []
+})
+export class ProfileLinkImageComponent implements OnInit {
+  @Input() username: string;
+
+  constructor(
+    private http: HttpClient,
```

Comment:
Please create a separate backend-api-service for the http request.

Reply:
Do we need to refactor the code too while performing Angular Migration? It'll take too long if code refactoring + Angular Migration go together. I really need to discuss this with other folks.

Reply:
I don't think the code should be refactored a lot. Just the http requests should be made using backend-api-services and they should return camelcase objects.

Reply:
Also, there may not be a large number of places where http requests are made?